sdk/search/Azure.Search.Documents/tests/Utilities/SharedSearchResources.cs

```diff
@@ -14,10 +15,38 @@ namespace Azure.Search.Documents.Tests
     [SetUpFixture]
     public class SharedSearchResources
     {
+        private static readonly AutoResetEvent s_event = new AutoResetEvent(true);
+        private static readonly TimeSpan s_timeout = TimeSpan.FromMinutes(1);
+
         /// <summary>
         /// The shared hotels search index.
         /// </summary>
-        public static SearchResources Search { get; set; }
+        public static SearchResources Search { get; private set; }
+
+        /// <summary>
+        /// Initializes the <see cref="Search"/> shared index if necessary.
+        /// </summary>
+        /// <param name="factory">The method to create the shared search index if necessary.</param>
+        /// <returns>A <see cref="ValueTask"/> that the method has completed.</returns>
+        public static async ValueTask EnsureInitialized(Func<Task<SearchResources>> factory)
```

Comment:
How would this work when two tests pass different initialization functions? Who wins?

Reply:
Doesn't really matter since the method is `EnsureInitialized`, using the same pattern/behavior as `LazyInitializer.EnsureInitialized`. The only winner is whoever sets the `Search` property first. How or what they do doesn't matter.

Reply:
It does matter because `LazyInitializer.EnsureInitialized` also takes a destination while you don't. So if two callers pass different lambdas that intend to create different resources they'll trample over each other.

Reply:
Fair point, but this is to initialize this particular shared instance of a class that would otherwise be static if not for supporting instantiation to support NUnit. This isn't a generic use case. The idea was to be a simple call so callers *don't* have to think about how to call this.

Reply:
Then why don't you move CreateWithHotelsIndexAsync method into SharedSearchResources.cs and expose SearchResources via a method like ValueTask<SearchResources> GetSearchResourcesAsync()

Reply:
Because it's also used outside that context as well. I don't see any value to change this as it's isolated for a very particular use case. It's target is very specific, and your argument would also apply if someone used `LazyInitializer.EnsureInitialized` with the same `ref` argument but different factory method as well.